### PR TITLE
Add Jitsi Meet Integration

### LINKED ISSUE #1445: Jitsi Meet
Add a Jitsi Meet integration, which enables people to use the public Jitsi meet instance (meet.jit.si) or use their own self hosted Jitsi.

Jitsi integration is as simple as generating a UUID and setting that as the URL (e.g. meet.jit.si/d8sf97s8d9f7s9).

### PR DESCRIPTION
**Pull request by philippdormann**
## What does this PR do?

This PR includes support for meet.jit.si Video Conferencing

Fixes #1445

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?
- Perform a booking with Jitsi Meet Video as the location 🙂

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PeerRich** [comment]
what's missing @philippdormann can we help?

**Comment by PeerRich** [comment]
> what's missing @philippdormann can we help?

ping

oh btw, "This branch is out-of-date with the base branch"

lets try and get this merged tomorrow 🎉

**Comment by philippdormann** [comment]
heya! it's done^^
figured you notices this and requested the reviewers 😅

just updated the branch :)

File: components/booking/pages/BookingPage.tsx [hunk 1]
```diff
@@ -195,7 +200,9 @@ const BookingPage = (props: BookingPageProps) => {
     })(),
   });
 
-  const getLocationValue = (booking: Pick<BookingFormValues, "locationType" | "phone">) => {
+  const getLocationValue = (
+    booking: Pick<BookingFormValues, "locationType" | "phone" | "integrations:jitsi">
+  ) => {
```
Comment (zomars) [inline components/booking/pages/BookingPage.tsx:205]: This is causing a type error and it seems unnecessary
```suggestion
  const getLocationValue = (booking: Pick<BookingFormValues, "locationType" | "phone">) => {
```

**Review by zomars — CHANGES_REQUESTED** [review]
We have some type issues in BookingPage and success page.

File: lib/integrations/getIntegrations.ts [hunk 2]
```diff
@@ -65,6 +66,14 @@ export const ALL_INTEGRATIONS = [
     description: "Video Conferencing",
     variant: "conferencing",
   },
+  {
+    installed: true,
```
[the remark this answers is not in the document]
Comment (philippdormann) [reply lib/integrations/getIntegrations.ts:70]: your decision - I think Jitsi Meet (and other services without signup) should be enabled by default.

File: lib/integrations/Jitsi/JitsiVideoApiAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    updateMeeting: (bookingRef: PartialReference): Promise<VideoCallData> => {
+      return Promise.resolve({
+        type: "jitsi_video",
+        id: bookingRef.meetingId as string,
+        password: bookingRef.meetingPassword as string,
+        url: bookingRef.meetingUrl as string,
+      });
+    },
+  };
```
[the remark this answers is not in the document]
Comment (philippdormann) [reply lib/integrations/Jitsi/JitsiVideoApiAdapter.ts:31]: not that i know of.

the huddle01 integration does the same thing.

File: pages/success.tsx [hunk 4]
```diff
@@ -141,7 +141,20 @@ export default function Success(props: inferSSRProps<typeof getServerSideProps>)
...
+                                    {location}
+                                  </a>
+                                </div>
+                              </>
+                            )}
+                            {!location.includes("http") && (
+                              <>
+                                <div className="col-span-2">{location}</div>
+                              </>
+                            )}
```
[the remark this answers is not in the document]
Comment (philippdormann) [reply pages/success.tsx:157]: ?

File: lib/integrations/getIntegrations.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zomars) [reply lib/integrations/getIntegrations.ts:66]: That's @PeerRich and @baileypumfleet decision to make. I'm just pointing it out.

File: pages/integrations/index.tsx [hunk 5]
```diff
@@ -487,6 +487,13 @@ function ConnectOrDisconnectIntegrationButton(props: {
       </div>
     );
   }
+  if (props.type === "jitsi_video") {
+    return (
+      <div className="px-3 py-2 truncate">
+        <h3 className="text-sm font-medium text-gray-700">{t("installed")}</h3>
+      </div>
+    );
+  }
```
[the remark this answers is not in the document]
Comment (zomars) [reply pages/integrations/index.tsx:496]: Can we apply these suggestions to avoid too much duplication?

**Review by zomars — COMMENTED** [review]
We're almost there. Just some suggestions pending.

File: pages/integrations/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (philippdormann) [reply pages/integrations/index.tsx]: done ✅

**Review by zomars — APPROVED** [review]

**Comment by philippdormann** [comment]
cheers, guys 🎉🎉

**Comment by baileypumfleet** [comment]
Thank you for your incredible work! We really appreciate the contribution :)

**Pull request merged**